### PR TITLE
feat: Add windows example ami

### PR DESCRIPTION
**Pull request by maths22**
This PR adds a packer config for building a Windows AMI based on 2019 core.
The packer config could presumably be reused to build on top of any other
Windows AMI, though I haven't tested it.  The example project for prebuilt
has been updated to allow using a windows AMI as well (mostly to show which
knobs need to be turned).

This has been tested against a personal aws account, but only with a super-
basic action that checks out the repo and does nothing else.

### CODE REVIEW

File: images/windows-core-2019/bootstrap_win.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Create a self-signed certificate to let ssl work
+$Cert = New-SelfSignedCertificate -CertstoreLocation Cert:\LocalMachine\My -DnsName "packer"
+New-Item -Path WSMan:\LocalHost\Listener -Transport HTTPS -Address * -CertificateThumbPrint $Cert.Thumbprint -Force
+
+# WinRM
+write-output "Setting up WinRM"
+write-host "(host) setting up WinRM"
+
+cmd.exe /c winrm quickconfig -q
```
Comment (ScottGuymer) [inline images/windows-core-2019/bootstrap_win.txt:22]: I think you can do this all in PowerShell but I can't find a good example of it being done non-interactively.

File: images/windows-core-2019/github_agent.windows.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+build {
+  name = "githubactions-runner"
+  sources = [
+    "source.amazon-ebs.githubrunner"
+  ]
+
+  provisioner "file" {
+    content = templatefile("../../modules/runners/templates/start-runner.ps1", {})
```
Comment (ScottGuymer) [inline images/windows-core-2019/github_agent.windows.pkr.hcl:55]: The `start-runner.ps1` from the runners template folder needs a wrapper script to tell it where to log its output.  When its used in the runners templates its included in the user-data script which handles the logging output.

You can look at start-runner.sh in the linux image for an example. Something like this

```powershell
Start-Transcript -Path "C:\UserData.log" -Append
${start_runner}
Stop-Transcript
```

File: images/windows-core-2019/github_agent.windows.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+source "amazon-ebs" "githubrunner" {
+  ami_name      = "github-runner-windows-core-2019-${formatdate("YYYYMMDDhhmm", timestamp())}"
+  communicator  = "winrm"
+  instance_type = "t3a.medium"
+  region        = var.region
+  source_ami_filter {
+    filters = {
+      name                = "Windows_Server-2019-English-Core-Base-*"
```
Comment (ScottGuymer) [inline images/windows-core-2019/github_agent.windows.pkr.hcl:29]: In the Linux image we include docker as standard, we should do this for windows too. We should either choose the core-containers base AMI or configure docker in the windows-provisioner.ps1.

**Comment by ScottGuymer** [comment]
Looks good.

Last thing I think needs to be added it to make the GH workflow a matrix job to run over the different images in the folder.

https://github.com/philips-labs/terraform-aws-github-runner/blob/develop/.github/workflows/packer-build.yml#L23

**Comment by ScottGuymer** [comment]
Looks like the build is failing on the formatting in `github_agent.windows.pkr.hcl`

**Comment by pbronneberg** [comment]
Great! looking forward to using the Windows Build agents

**Review by ScottGuymer — APPROVED** [review]
LGTM

**Comment by ScottGuymer** [comment]
@npalm anything else to add?

**Review by npalm — APPROVED** [review]
LGTM

**Pull request merged**